### PR TITLE
Trait:vegeta

### PR DESCRIPTION
**Pull request by oeular**
Signed-off-by: Xueliang Yang <yangxueliang@cmbchina.com>

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` or `[Trait]`).
- [x] Updated/Added any relevant [documentation] and [examples].
- [x] Unit/E2E Tests added.

### CODE REVIEW

File: addons/vegeta/readme.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          properties:
+            name: show-vegeta
+            domain: http://show-vegeta:80
+            parallelism: 1
+            backofflimit: 10
+```
+
+- Check the app status and trait status
+```shell
+kubectl get app -n vegeta
```
Comment (wonderflow) [inline addons/vegeta/readme.md:36]: let's use vela command for all the things instead of kubectl

**Review by wangyikewxgm — COMMENTED** [review]
testing by creating a app with this trait but got an error:
```yaml
 message: 'step apply: run step(provider=oam,do=component-apply): GenerateComponentManifest:
        evaluate trait=vegeta template for component=show-vegeta app=show-vegeta:
        failed to have the workload/trait unstructured: spec.template.spec.containers.args:
        reference "parameter" not found (and 1 more errors)'
```

**Comment by oeular** [comment]
@wangyikewxgm  Thanks so much. I update the README.md.

File: addons/vegeta/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: vegeta
+version: 12.8.4
+description: Vegeta is a versatile HTTP load testing tool built out of a need to drill HTTP services with a constant request rate. It can be used both as a command line utility and a library.
+url: https://github.com/tsenart/vegeta
+
```
Comment (wangyikewxgm) [inline addons/vegeta/metadata.yaml:5]: can we have a icon for this addon?

File: addons/vegeta/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+tags:
+  - extended_trait
+
+deployTo:
+  control_plane: true
+  runtime_cluster: true
```
[the remark this answers is not in the document]
Comment (oeular) [reply addons/vegeta/metadata.yaml:11]: Nice review.

File: addons/vegeta/metadata.yaml (hunk 2 above)
Comment (oeular) [reply addons/vegeta/metadata.yaml:5]: ok

File: addons/vegeta/readme.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+
+# Supported workload type
+Vegeta Trait supports following component types: webservice、worker and cloneset.
```
Comment (wangyikewxgm) [inline addons/vegeta/readme.md:3]: why only support  tree types component?

Comment (oeular) [reply addons/vegeta/readme.md:3]: Not only. The three types are easy to use.

Comment (oeular) [reply addons/vegeta/readme.md:3]: Other types are alone to users to try.

Comment (wangyikewxgm) [reply addons/vegeta/readme.md:3]: So we can remove this notice? since the `appliesToWorkloads` can take effect to all workloads.

Comment (wonderflow) [reply addons/vegeta/readme.md:3]: I don't think it works for all workloads. We must make it accurate.

**Review by wonderflow — DISMISSED** [review]

File: addons/vegeta/readme.md (hunk 4 above)
Comment (wangyikewxgm) [reply addons/vegeta/readme.md:3]: So `appliesToWorkloads` filed should be align with this notice.

**Review by wonderflow — APPROVED** [review]

**Review by wangyikewxgm — APPROVED** [review]
Good job!

**Pull request merged**